### ISSUE TITLE
`logback-test.xml` in the Main Avro Resources Folder Can Cause Runtime Issues for Applications

### ISSUE THREAD

**Issue by kyle-silver** [FasterXML/jackson-dataformats-binary#379]
Because this test logback file (located [here](https://github.com/FasterXML/jackson-dataformats-binary/blob/2.15/avro/src/main/resources/logback-test.xml)) is in the `main` directory, it gets packaged and deployed as part of the release artifact. I have some apps running with Spring Boot 2.7.x and Logback and have observed that this test configuration file is getting picked up by Logback and causing the app to fail on startup due to a (perceived) misconfiguration. I believe this can be remediated by moving `logback-test.xml` to the `avro/test/resources` directory.

---

I am happy to open a pull request with this change myself but wanted to file an issue first and hear feedback from the maintainers. I am a huge fan of the work you do on this library

**Comment by cowtowncoder** [comment]
Sounds like a flaw, unintended so PR (against 2.14 branch so I can merge forward) would be most welcome.
Thank you for reporting this.

**Comment by kyle-silver** [comment]
Absolutely. One question, though—since that file doesn't exist on the 2.14 branch, will this need a separate commit on the 2.15 branch to remove it once the test file is added?

**Comment by cowtowncoder** [comment]
Oh ok then just do that for 2.15 branch. I haven't had a chance to check (traveling this week).

**Comment by techabstraction** [comment]
Hope you don't mind me raising this PR with reaching out first. Just trying to help out, as we're also experiencing our apps failing on start up.

**Comment by cowtowncoder** [comment]
@techabstraction Not all, thanks for PR! I wish I had remembered to wait for this before 2.15.1 release, but it'll be in 2.15.2.

**Comment by techabstraction** [comment]
Thanks @cowtowncoder . Any idea when we can expect `2.15.2`? `2.14.3` has a high priority CVE so isn't safe to use. 

```
PRISMA-2023-0067

Priority: High

com.fasterxml.jackson.core_jackson-core package versions before 2.15.0 are vulnerable to Denial of Service (DoS). The package does not properly restrict the size or amount of resources that are requested or influenced by an actor, which can be used to consume more resources than intended and leads to Uncontrolled Resource Consumption ('Resource Exhaustion').
```

**Comment by techabstraction** [comment]
Hi @cowtowncoder . Just checking in, any update on when `2.15.2` will be released?

**Comment by cowtowncoder** [comment]
This week, today or tomorrow I hope.

**Comment by SuvaAndiBill** [comment]
Hi,
just to let you know: Also with spring-boot 3.x.x the app fails to start with version 2.15.1.
With version 2.15.2 it's ok again.
Thx for your work.

**Comment by cowtowncoder** [comment]
Thank you for verifying @SuvaAndiBill -- too bad this file was included but good things work again.
